tests/test_fillnodata.py

```diff
@@ -19,13 +21,27 @@ def test_fillnodata():
     result = fillnodata(a, mask)
     assert(np.all((np.ones([3, 3]) * 42) == result))
 
+
+def test_fillnodata_masked_array():
+    """Test filling nodata values in a masked ndarray"""
+    # create a 5x5 array, with some missing data
+    a = np.ones([3, 3]) * 42
```

Comment:
For consistency, I'd recommend using then altering one of the existing test fixtures, e.g., [`basic_image`](https://github.com/mapbox/rasterio/blob/master/tests/conftest.py#L105)
or [`pixelated_image`](https://github.com/mapbox/rasterio/blob/master/tests/conftest.py#L143)